temporal-sdk/src/main/java/io/temporal/internal/client/WorkflowClientInternal.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.internal.client;
+
+import io.temporal.client.WorkflowClient;
+import io.temporal.worker.WorkerFactory;
+
+/**
+ * From OOP point of view, there is no reason for this interface to don't extend {@link
+ * WorkflowClient}. It's not extending only to make sure that the calls that can come through a
+ * user-supplied WorkflowClient are made through that entity whenever possible instead of the entity
+ * obtained by {@link WorkflowClient#getInternal()}. This will make sure DI/AOP frameworks or user's
+ * wrapping code implemented in WorkflowClient proxy or adapter gets executed when possible and
+ * {@link WorkflowClient#getInternal()} is used only for internal functionality.
+ */
+public interface WorkflowClientInternal {
```

Comment:
If you want, I'd support just removing this interface and casting directly to the internal class impl. But I understand if you feel better about the abstraction.

Reply:
I envision that some very advanced and invasive user may want to intercept getInternal to create a wrapper for logging/tracing purposes. It's not something we may ever recommend, but some users may want it anyway. It will be cleaner to have an interface for them.
But I don't have a strong opinion on this one and can replace it with a class usage, I see a value in actually PREVENTING the above scenario in doing this too.

Reply:
I think I'd just prevent that for now and if anyone ever makes a fuss consider it then. Seems like a really edge situation to me.

Reply:
Nah, after second consideration, this solution would require opening WorkflowClientInternal implementation to `public`. I think it's cleaner to open only an internal interface than the whole implementation class.